cmd/funcbench/main.go

```diff
@@ -0,0 +1,261 @@
+		if err != nil {
+			log.Println(string(data))
+			log.Fatalln(err)
+		}
+
+		cmd := exec.Command("git", "merge", "--squash", "--no-commit", "pullrequest")
+		data, err = cmd.CombinedOutput()
+		if err != nil {
+			log.Println(string(data))
+			log.Fatalln(err)
+		}
+		if cmd.ProcessState.ExitCode() != 0 {
+			if err := postComment(clt, owner, repo, prNumber, "Git merge failed"); err != nil {
+				log.Fatalln(err)
+			}
+		}
+
+		data, err = exec.Command("git", "reset").CombinedOutput()
+		if err != nil {
+			log.Println(string(data))
+			log.Fatalln(err)
+		}
+
+		// Benchmark the with pullrequest changes.
+		if race == "-no-race" {
+			cmd = exec.Command("go", "test", "-bench", issueComment, "-benchmem", "-v", "./...")
+		} else {
+			cmd = exec.Command("go", "test", "-bench", issueComment, "-benchmem", "-race", "-v", "./...")
+		}
+		data, err = cmd.CombinedOutput()
+		log.Println(string(data))
+		if err != nil || cmd.ProcessState.ExitCode() != 0 {
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Go test for this pull request failed. Check https://github.com/%s/%s/commit/%s/checks for more information.", owner, repo, latestCommitHash)); err != nil {
+				log.Fatalln(err)
+			}
+			log.Fatalln(err)
+		}
+		err = ioutil.WriteFile(newPath, data, 0644)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Checkout to the comparing branch.
+		filepath.Walk(os.Getenv("GITHUB_WORKSPACE"), func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() || strings.Contains(path, ".git/") || strings.HasSuffix(info.Name(), "test.go") {
+				return nil
+			}
+
+			data, err = exec.Command("git", "checkout", "--", path).CombinedOutput()
+			if err != nil {
+				log.Println(string(data))
+				log.Fatalln(err)
+			}
+			return nil
+		})
+
+		// Benchmark the comparing branch.
+		if race == "-no-race" {
+			cmd = exec.Command("go", "test", "-bench", issueComment, "-benchmem", "-v", "./...")
+		} else {
+			cmd = exec.Command("go", "test", "-bench", issueComment, "-benchmem", "-race", "-v", "./...")
+		}
+		data, err = cmd.CombinedOutput()
+		log.Println(string(data))
+		if err != nil || cmd.ProcessState.ExitCode() != 0 {
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Go test on branch %s failed. Check https://github.com/%s/%s/commit/%s/checks for more information.", branch, owner, repo, latestCommitHash)); err != nil {
+				log.Fatalln(err)
+			}
+			log.Fatalln(err)
+		}
+		err = ioutil.WriteFile(oldPath, data, 0644)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		data, err = exec.Command(strings.Join([]string{os.Getenv("GOPATH"), "/bin/benchcmp"}, ""), oldPath, newPath).CombinedOutput()
+		if err != nil {
+			log.Println(string(data))
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Error: `benchcmp` failed. Check https://github.com/%s/%s/commit/%s/checks for more information.", owner, repo, latestCommitHash)); err != nil {
```

Comment:
Best to put this in some var instead of repeating it so many times
logsLink := fmt.Sprintf("https://github.com/%s/%s/commit/%s/checks for more information.", owner, repo, latestCommitHash)